src/pages/quiz-page.tsx

```diff
@@ -0,0 +1,196 @@
+import { toast } from 'react-toastify';
+import {
+  Button, ButtonProps, Container, Form, Header, Radio,
+} from 'semantic-ui-react';
+import {
+  MESSAGE_ANSWER_CORRECT,
+  MESSAGE_ANSWER_WRONG,
+  BUTTON_END_QUIZ_TEXT,
+  BUTTON_NEXT_TEXT,
+  BUTTON_SKIP_TEXT,
+  BUTTON_SUBMIT_TEXT,
+  TOAST_SKIP_QUESTION,
+  LEVEL,
+  SCORE,
+  toastConfig,
+  BUTTON_DISABLE_CLASS_NAME,
+  BUTTON_ENABLE_CLASS_NAME,
+  RADIO_CORRECT_COLOR,
+  RADIO_WRONG_COLOR,
+  QUESTION,
+  BUTTON_END_QUIZ_COLOR,
+  FOOTER_MARGIN,
+  BUTTON_WIDTH_DEFAULT,
+  FONT_SIZE_MEDIUM,
+} from '../contants';
+
+const styles = {
+  containerFlex: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    flexDirection: 'column',
+    height: '80vh',
+  },
+  containerHeader: {
+    flex: '1',
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    height: '100%',
+    fontSize: FONT_SIZE_MEDIUM,
+    fontWeight: 'bold',
+  },
+  containerQuiz: {
+    flex: '3',
+    display: 'flex',
+    justifyContent: 'center',
+    flexDirection: 'column',
+  },
+  containerFooter: {
+    flex: '1',
+    display: 'flex',
+    justifyContent: 'center',
+    flexDirection: 'column',
+  },
+  containerButtons: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    height: '100%',
+  },
+  radio: {
+    textAlign: 'left',
+    width: '100%',
+  },
+  buttonWrapper: {
+    display: 'flex',
+    justifyContent: 'center',
+  },
+  button: {
+    width: BUTTON_WIDTH_DEFAULT,
+  },
+};
+
+const QuizPage: React.FC = (): ReactElement => {
+  /**
+   * Normally this should be passed from props. This will be removed once we connected with other components..
+   */
+  const props = {
+    levelOfDifficulty: 1,
+  };
+
+  /**
+   * This mock data will be removed once the API endpoint is built.
+   */
+  const requestData = {
+    question: 'Which company belongs to 31st Engineer Battalion?',
+    choices: ['Smoking Aces', 'Dawgs', 'Cobras', 'Easy Company'],
+    correctChoiceIndex: 2,
+  };
+
+  const [score, setScore] = useState(0);
+  const [questionNumber, setQuestionNumber] = useState(1);
+  const [indexOfSelectedRadio, setIndexOfSelectedRadio] = useState(-1);
+  const [hasChooseAnswer, setHasChooseAnswer] = useState(false);
+  const [hasSubmitAnswer, setHasSubmitAnswer] = useState(false);
+
+  const isCorrectAnswer = (indexOfRadio: number) => indexOfRadio === requestData.correctChoiceIndex;
+
+  const getRadioResultColor = (index: number) => (isCorrectAnswer(index) ? RADIO_CORRECT_COLOR : RADIO_WRONG_COLOR);
+
+  const goToNextQuestion = () => {
+    setIndexOfSelectedRadio(-1);
+    setHasChooseAnswer(false);
+    setHasSubmitAnswer(false);
+    setQuestionNumber(questionNumber + 1);
+  };
+
+  const onClickRadio = (event: React.MouseEvent, { index }: ButtonProps) => {
+    if (!hasSubmitAnswer) {
+      setIndexOfSelectedRadio(index);
+      setHasChooseAnswer(true);
+    }
+  };
+
+  const onClickSkip = () => {
```

Comment:
If a question is skipped, what is the expectation on the backend? Is it treated as an incorrect answer?

Reply:
We can treat it in either way. Depends on the requirement.
`onClickSkip` and `onClickSubmit` are incomplete since API endpoints are not built yet. I will re-confirm this with you once the server is deployed.